From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 17 Mar 2014 23:51:09 +0000
Subject: fs: Avoid ABI change in 3.13.6
Forwarded: not-needed

The second parameter to sync_inodes_sb() was removed.  Add it back
but ignore the value.

diff --git a/fs/fs-writeback.c b/fs/fs-writeback.c
index d754e3c..d9c0c0c 100644
--- a/fs/fs-writeback.c
+++ b/fs/fs-writeback.c
@@ -1349,12 +1349,13 @@ EXPORT_SYMBOL(try_to_writeback_inodes_sb);
 
 /**
  * sync_inodes_sb	-	sync sb inode pages
- * @sb: the superblock
+ * @sb:			the superblock
+ * @older_than_this:	dummy parameter for binary compatibility
  *
  * This function writes and waits on any dirty inode belonging to this
  * super_block.
  */
-void sync_inodes_sb(struct super_block *sb)
+void sync_inodes_sb(struct super_block *sb, unsigned long older_than_this)
 {
 	DECLARE_COMPLETION_ONSTACK(done);
 	struct wb_writeback_work work = {
diff --git a/fs/sync.c b/fs/sync.c
index 6c0ca3b..5eb0185 100644
--- a/fs/sync.c
+++ b/fs/sync.c
@@ -30,7 +30,7 @@
 static int __sync_filesystem(struct super_block *sb, int wait)
 {
 	if (wait)
-		sync_inodes_sb(sb);
+		sync_inodes_sb(sb, 0);
 	else
 		writeback_inodes_sb(sb, WB_REASON_SYNC);
 
@@ -70,7 +70,7 @@ EXPORT_SYMBOL_GPL(sync_filesystem);
 static void sync_inodes_one_sb(struct super_block *sb, void *arg)
 {
 	if (!(sb->s_flags & MS_RDONLY))
-		sync_inodes_sb(sb);
+		sync_inodes_sb(sb, 0);
 }
 
 static void sync_fs_one_sb(struct super_block *sb, void *arg)
diff --git a/fs/xfs/xfs_super.c b/fs/xfs/xfs_super.c
index d971f49..f317488 100644
--- a/fs/xfs/xfs_super.c
+++ b/fs/xfs/xfs_super.c
@@ -913,7 +913,7 @@ xfs_flush_inodes(
 	struct super_block	*sb = mp->m_super;
 
 	if (down_read_trylock(&sb->s_umount)) {
-		sync_inodes_sb(sb);
+		sync_inodes_sb(sb, jiffies);
 		up_read(&sb->s_umount);
 	}
 }
diff --git a/include/linux/writeback.h b/include/linux/writeback.h
index 021b8a3..fc0e432 100644
--- a/include/linux/writeback.h
+++ b/include/linux/writeback.h
@@ -97,7 +97,7 @@ void writeback_inodes_sb_nr(struct super_block *, unsigned long nr,
 int try_to_writeback_inodes_sb(struct super_block *, enum wb_reason reason);
 int try_to_writeback_inodes_sb_nr(struct super_block *, unsigned long nr,
 				  enum wb_reason reason);
-void sync_inodes_sb(struct super_block *);
+void sync_inodes_sb(struct super_block *sb, unsigned long older_than_this);
 void wakeup_flusher_threads(long nr_pages, enum wb_reason reason);
 void inode_wait_for_writeback(struct inode *inode);
 
